### PR TITLE
Update files for all boards 

### PR DESCRIPTION
**Pull request by kolerov**
* Use common linker script and .x memory map file for boards. It
  gets reed of putting memory.x file in build directory. You just
  need to use an appropriate specs file to set vector base,
  include a corresponding `memory.x` file and link with UART
  library.
* Add support of EM SDP 1.1 ad 1.2.

### CODE REVIEW

File: libgloss/arc/emsdp1.1.specs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+%rename link emsdp_link
+%rename link_gcc_c_sequence emsdp_link_gcc_c_sequence
+%rename startfile emsdp_startfile
+
+*link:
+%(emsdp_link) -T emsdp1.1.x%s -T arcv2elf-common.ld%s
```
Comment (claziss) [inline libgloss/arc/emsdp1.1.specs:6]: You don't need to have two linker scripts here, just use the `emsdp1.1.x`  and include the common one in this file.

**Comment by kolerov** [comment]
@claziss I've updated PR and renamed `.x` to `.ld` since it's technically linker scripts.

File: libgloss/arc/emsdp1.1.ld [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    ICCM0 : ORIGIN = 0x60000000, LENGTH = 128K
+    DCCM  : ORIGIN = 0x80000000, LENGTH = 128K
+}
+
+REGION_ALIAS("startup", ICCM0)
+REGION_ALIAS("text", ICCM0)
+REGION_ALIAS("data", DCCM)
+REGION_ALIAS("sdata", DCCM)
+
+PROVIDE (__stack_top = (0x8001ffff & -4));
```
Comment (claziss) [inline libgloss/arc/emsdp1.1.ld:15]: You don't need `__stack_top` and `__end_heap` as they are defined in the common part ;)

**Review by claziss — CHANGES_REQUESTED** [review]
Please remove the unneeded linker variables.

File: libgloss/arc/emsdp1.1.ld (hunk 2 above)
Comment (kolerov) [reply libgloss/arc/emsdp1.1.ld]: Oops! Indeed.

**Review by claziss — APPROVED** [review]

**Pull request merged**